subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/billing/PlayBillingManager.kt

```diff
@@ -348,6 +364,36 @@ class RealPlayBillingManager @Inject constructor(
             }
         }
     }
+
+    private suspend fun loadPurchases() {
+        when (val result = billingClient.queryPurchases()) {
+            is QueryPurchasesResult.Success -> {
+                purchases = result.purchases
+                logcat { "Billing: Loaded ${result.purchases.size} active purchases" }
+            }
+            is QueryPurchasesResult.Failure -> {
+                logcat { "Billing: Failed to load purchases: ${result.billingError} - ${result.debugMessage}" }
+            }
+        }
+    }
+
+    override suspend fun getLatestPurchaseToken(): String? = withContext(dispatcherProvider.io()) {
```

Comment:
nit: Unless I'm missing something, this method doesn't do any blocking I/O operations, so switching dispatcher is unnecessary. You could even remove `suspend` from this method.

Reply:
Good catch! I'll remove the suspend modifier too